client/src/lib/solana/sol-data.ts

```diff
@@ -287,29 +357,82 @@ export class ServiceEndpoint extends Assignable {
     };
   }
 
-  static parse(service: DIDServiceEndpoint): ServiceEndpoint {
+  static async parse(service: DIDServiceEndpoint): Promise<ServiceEndpoint> {
     return new ServiceEndpoint({
-      id: DecentralizedIdentifier.parse(service.id).urlField,
+      id: (await DecentralizedIdentifier.parse(service.id)).urlField,
       endpointType: service.type,
       endpoint: service.serviceEndpoint,
       description: service.description,
     });
   }
 }
 
+/**
+ * A class representing a SOL Did
+ */
 export class DecentralizedIdentifier extends Assignable {
+  /**
+   * The cluster the DID points to
+   */
   clusterType: ClusterType;
-  pubkey: SolPublicKey;
-  urlField: string;
+  /**
+   * The address of the DID
+   */
+  authorityPubkey: SolPublicKey;
+  /**
+   * The key to the DID data
+   */
+  pdaPubkey: SolPublicKey;
+  /**
+   * The optional field following the DID address and `#`
+   */
+  urlField?: string;
+
+  /**
+   * Creates a new `DecentralizedIdentifier` from its requisite parts.
+   *
+   * Use `DecentralizedIdentifier::parse` to obtain this from a direct did address.
+   *
+   * @param clusterType
+   * @param authorityPubkey
+   * @param urlField
+   * @param pdaPubkey
+   */
+  constructor({
+    clusterType,
+    authorityPubkey,
+    urlField,
+    pdaPubkey,
+  }: {
```

Comment:
As above, consider moving to a type. SInce the type is essentially typeof DecentralizedIdentifier here I am wondering if there is a better way to use Borsh? Seems like a lot of boilerplate.